[API V2] Client Library and Doc Update - PHP #680

paulorbpinho-fullstacklabs opened the pull request:
Description:
Created CurrencyPercentageTier and PercentageTier resources.
Created test cases to new resources.
Updated test cases of Addon and Subscription.

mwlang commented on Tests/Recurly/Addon_Test.php:
```diff
@@ -59,4 +59,26 @@ public function testCreateXmlTieredAddOn() {
+
+    $this->assertEquals("<?xml version=\"1.0\"?>\n<add_on><add_on_code>little_llama</add_on_code><tier_type>tiered</tier_type><percentage_tiers><percentage_tier><currency>USD</currency><tiers><tier><ending_amount_in_cents>100</ending_amount_in_cents><usage_percentage>10.0</usage_percentage></tier><tier><ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents><usage_percentage>20.0</usage_perc ...
+  }
 }
```
I’m curious, it it possible (and easy) to format this to be more readable?
```php
$this->assertEquals("<?xml version=\"1.0\"?>\n<add_on><add_on_code>little_llama</add_on_code><tier_type>tiered</tier_type><percentage_tiers><percentage_tier><currency>USD</currency><tiers><tier><ending_amount_in_cents>100</ending_amount_in_cents><usage_percentage>10.0</usage_percentage></tier><tier><ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents><usage_percentage>20.0</usage_percentage></tier></tiers></percentage_tier></percentage_tiers></add_on>\n", $addon->xml());
```

something like:
```php
$this->assertEquals("
  <?xml version=\"1.0\"?>\n<add_on>
    <add_on_code>little_llama</add_on_code>
    <tier_type>tiered</tier_type>
    <percentage_tiers>
      <percentage_tier>
        <currency>USD</currency>
        <tiers>
          <tier>
            <ending_amount_in_cents>100</ending_amount_in_cents>
            <usage_percentage>10.0</usage_percentage>
          </tier>
          <tier>
            <ending_amount_in_cents nil=\"nil\"></ending_amount_in_cents>
            <usage_percentage>20.0</usage_percentage>
          </tier>
        </tiers>
      </percentage_tier>
    </percentage_tiers>
  </add_on>
", $addon->xml());
```

paulorbpinho-fullstacklabs replied:
Yes is possible if we change assertEquals to assertXmlStringEqualsXmlString. I changed that.

mwlang commented on lib/recurly/percentage_tier_by_currency.php:
```diff
@@ -0,0 +1,41 @@
+ * class Recurly_PercentageTierByCurrency
+ * @property float $ending_amount_in_cents Ending amount for the tier. Must be left empty if it is the final tier.
+ * @property string $usage_percentage The percentage taken of the monetary amount of usage tracked. This can be up to 4 decimal places represented as a string.
+ */
```
**question**: If we wanted 5.25%, is that passed as "5.25" or "0.0525"?  It is perhaps worth documenting more clearly here.  Example:
```php
/**
* @property string $usage_percentage The percentage taken of the monetary amount of usage tracked. This can be up to 4 decimal places represented as a string (e.g. 5.25% is passed as "0.0525")
* */

paulorbpinho-fullstacklabs replied:
Changed

paulorbpinho-fullstacklabs committed:
[API V2] Client Library and Doc Update - PHP

Description:
Created CurrencyPercentageTier and PercentageTier resources.
Created test cases to new resources.
Updated test cases of Addon and Subscription.

mwlang reviewed (approved):
This looks solid to me.  The reformatted XML is definitely a readability improvement in the test specs.

JJDrill commented:
+1 QA
